src/spxRender.js

```diff
@@ -78,11 +84,64 @@ function renderMixedTextToString(text, suppressWarnings) {
             }
         } else {
             bit = htmlEscape(bit.replace("\\$", "$"));
-            // And now this function is terribly named...
-            bits[i] = preprocessText(bit);
+
+            // Replace Latex tags such as \textbf{...} and \textit{...} with
+            // HTML tags such as <b>...</b> and <i>...</i>
+            bit = processLatex(bit, tagStack);
+            bits[i] = bit;
         }
     }
-    return bits.join('');
+    return preprocessText(bits.join(''));
+}
+
+/**
+ * Replaces Latex sections of a bit of a string with equivalent HTML elements.
+ * At present \textbf{ ... } is replaced with <b> ... </b> and |textit{ ... }
+ * is replaced with <i> ... </i>
+ * @param bit The bit of the string to be modified
+ * @param tagStack A stack of the open tags in the string (to allow a tag
+ * opened in one bit to be closed in a following bit, preserving the
+ * opening/closing order)
+ * @returns {*} The modified bit of a string
+ */
+function processLatex(bit, tagStack) {
+    var firstTagIdx, firstTag, index, tag, lastOpenedTag;
+
+    // Find the first Latex tag or closing brace in the bit. Replace a Latex
+    // tag its corresponding HTML tag, and then push the tag onto the stack
+    // to be closed when we get to the next closing brace (which might be in
+    // the next bit). If we encounter a closing brace, replace it with the
+    // top closing tag on the stack. Continue repeating until there are no more
+    // Latex tags or closing braces in the string.
+    do {
+        firstTagIdx = -1;
+        firstTag = null;
+        index = -1;
+        for (tag in latexTags) {
+            index = bit.indexOf(tag);
+            if (index !== -1 && (index < firstTagIdx || firstTagIdx === -1)) {
+                firstTagIdx = index;
+                firstTag = tag;
+            }
+        }
+        index = bit.indexOf("}");
+        if (index !== -1 && (index < firstTagIdx || firstTagIdx === -1)) {
```

Comment:
This is the same as the for loop above - might be worth at least commenting it as such